test/unit/StakingPool/requestAllocation.js

```diff
@@ -1530,4 +1530,117 @@ describe('requestAllocation', function () {
+    expect(midAllocations).to.be.deep.equal(capacities);
+
+    const tokenId = await stakingNFT.totalSupply();
+    await stakingPool.connect(staker).extendDeposit(tokenId, stakeTrancheId, stakeTrancheId + 1, 0);
+
+    const unfullfillableRequest = {
+      ...allocationRequestParams,
+      // targetting tranche idx 5
+      period: daysToSeconds(91 * 4),
+      gracePeriod: daysToSeconds(91),
+    };
+
+    await expect(
+      stakingPool.connect(this.coverSigner).requestAllocation(
+        NXM_PER_ALLOCATION_UNIT, // smallest amount
+        previousPremium,
+        unfullfillableRequest,
+      ),
+    ).to.be.revertedWithCustomError(stakingPool, 'InsufficientCapacity');
+  });
+
+  it('accounts for carried over allocations partially filling the capacity', async function () {
+    const { stakingPool, stakingNFT } = this;
+    const [staker] = this.accounts.members;
+    const { NXM_PER_ALLOCATION_UNIT } = this.config;
+
+    const { productId } = allocationRequestParams;
+    const firstCoverAmount = parseEther('80000');
+    const secondCoverAmount = parseEther('20000');
+    const previousPremium = 0;
+
+    const allocationRequest = {
+      ...allocationRequestParams,
+      period: daysToSeconds(10),
+      gracePeriod: 0,
+    };
+
+    const currentTrancheId = await moveTimeToNextTranche(1);
+    const stakeTrancheId = currentTrancheId + trancheOffset - 1;
+
+    const initialAllocations = await stakingPool.getActiveAllocations(productId);
+    initialAllocations.forEach(allocation => {
+      expect(allocation).to.be.equal(0);
+    });
+
+    // allocate all available capacity
+    await stakingPool.connect(this.coverSigner).requestAllocation(firstCoverAmount, previousPremium, allocationRequest);
+
+    // expect all available capacity to be used
+    const midAllocations = await stakingPool.getActiveAllocations(productId);
+    const expectedTrancheAllocation = firstCoverAmount.div(NXM_PER_ALLOCATION_UNIT);
+    const expectedAllocations = [Zero, Zero, Zero, Zero, expectedTrancheAllocation, Zero, Zero, Zero];
+    expect(midAllocations).to.be.deep.equal(expectedAllocations);
+
+    const tokenId = await stakingNFT.totalSupply();
+    await stakingPool.connect(staker).extendDeposit(tokenId, stakeTrancheId, stakeTrancheId + 1, 0);
+
+    const unfullfillableRequest = {
+      ...allocationRequestParams,
+      // targetting tranche idx 5
+      period: daysToSeconds(91 * 4),
+      gracePeriod: daysToSeconds(91),
+    };
+
+    await expect(
+      stakingPool.connect(this.coverSigner).requestAllocation(
+        secondCoverAmount.add(1), // slightly over the limit
+        previousPremium,
+        unfullfillableRequest,
+      ),
+    ).to.be.revertedWithCustomError(stakingPool, 'InsufficientCapacity');
+
+    stakingPool.connect(this.coverSigner).requestAllocation(
+      secondCoverAmount, // exact available amount
+      previousPremium,
+      unfullfillableRequest,
```

Comment:
this one is actually fulfillable right?

it doesn't revert

Reply:
Adjusted the variable names.